src/main/kotlin/com/csbroker/apiserver/repository/OAuth2AuthorizationRequestBasedOnCookieRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.csbroker.apiserver.repository
+
+import com.csbroker.apiserver.common.util.addCookie
+import com.csbroker.apiserver.common.util.deleteCookie
+import com.csbroker.apiserver.common.util.deserialize
+import com.csbroker.apiserver.common.util.getCookie
+import com.csbroker.apiserver.common.util.serialize
+import org.springframework.security.oauth2.client.web.AuthorizationRequestRepository
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+const val OAUTH2_AUTHORIZATION_REQUEST_COOKIE_NAME: String = "oauth2_auth_request"
+const val REDIRECT_URI_PARAM_COOKIE_NAME: String = "redirect_uri"
+const val REFRESH_TOKEN: String = "refresh_token"
+private const val cookieExpireSeconds = 180
+
+class OAuth2AuthorizationRequestBasedOnCookieRepository : AuthorizationRequestRepository<OAuth2AuthorizationRequest> {
+    override fun loadAuthorizationRequest(request: HttpServletRequest): OAuth2AuthorizationRequest? {
+        val cookie = getCookie(request, OAUTH2_AUTHORIZATION_REQUEST_COOKIE_NAME) ?: return null
+
+        return deserialize(cookie, OAuth2AuthorizationRequest::class.java)
+    }
+
+    override fun saveAuthorizationRequest(
+        authorizationRequest: OAuth2AuthorizationRequest?,
+        request: HttpServletRequest,
+        response: HttpServletResponse
+    ) {
+        if (authorizationRequest == null) {
+            deleteCookie(request, response, OAUTH2_AUTHORIZATION_REQUEST_COOKIE_NAME)
+            deleteCookie(request, response, REDIRECT_URI_PARAM_COOKIE_NAME)
+            deleteCookie(request, response, REFRESH_TOKEN)
+            return
+        }
+
+        addCookie(
+            response, OAUTH2_AUTHORIZATION_REQUEST_COOKIE_NAME, serialize(authorizationRequest), cookieExpireSeconds
```

Comment:
🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)